### ISSUE TITLE
Request Object getRequestTarget

### ISSUE THREAD

**Issue by geggleto** [slimphp/Slim#1587]
In RC2 - RC1

```
//URL http://localhost/video-site/public/test

$app->get('/test', function ($req, $res, $args = []) {
   print $req->getRequestTarget();
});
```

Acutal: /video-site/publictest
Expected: /video-site/public/test

**Comment by RyanNerd** [comment]
You should have gotten a 404 error. Since your URI path of /video-site/public/test isn't a registered route.
Your code registers a get path of '/test' not '/video-site'

``` PHP
$app->get('/test', function ($req, $res, $args = []) {
   print $req->getRequestTarget();
});
```

Only routes that **start** with /test would be captured with the `getRequestTarget()` logic.

**Comment by tuupola** [comment]
@geggleto most likely has Slim installed in `/video-site/public/` folder. Looking at [the code](https://github.com/slimphp/Slim/blob/3.x/Slim/Http/Request.php#L458) cause is similar as with issue #1554. `$this->uri->getPath()` returns path without leading slash in subfolder, so I guess `$this->uri->getBasePath()` returns rest of the path without trailing slash?

**Comment by geggleto** [comment]
@tuupola  yeah I have it installed to that directory :) The getBasePath does return the correct path however I was trying to decouple away from Slim's non PSR-7 methods.

**Comment by geggleto** [comment]
A possible solution might be simply inspecting the REQUEST_URI attribute of the server envrionment.

```
$req->getServerParams()['REQUEST_URI']
```

**Comment by akrabat** [comment]
I can reproduce this. 

<s>I think the correct fix is to ensure that the basePath always ends with a slash if it's not empty.</s>

Actually, looking at Uri::__toString(), we need to add the '/' in `getRequestTarget()`...

**Comment by akrabat** [comment]
Fixed in #1633
